pkg/apis/intevents/v1alpha1/brokercell_types.go

```diff
@@ -68,9 +68,33 @@ var (
 	_ duckv1.KRShaped = (*BrokerCell)(nil)
 )
 
+// SystemResource specifies quantities for resources such as CPU and memory
+type SystemResource struct {
+	CPU    *string `json:"cpu,omitempty"`
+	Memory *string `json:"memory,omitempty"`
+}
+
+// ResourceSpecification defines requests and limits for the system resources
+type ResourceSpecification struct {
+	// Request specifies the minimal amount of the resource for the deployment to be schedulable
+	Requests SystemResource `json:"requests,omitempty"`
+
+	// Limit specifies the maximal amount of the resource that the deployment is allowed to consume
+	Limits SystemResource `json:"limits,omitempty"`
+}
```

Comment:
Sorry if I'm appearing to be back and forth. I had a second thought on the defaulting. I found the current logic could still be confusing. E.g. If I (as a user) set the CPU requirements but not memory, then it will use the default memory requirements. But such defaulting could somehow "invalidate" my CPU requirements as the memory defaults could be much lower than my expectation.

What if we do it with a more blunt approach? Either you (as a user) don't specify the `ComponentParameters` and we default everything for you, or you need to configure the whole `ComponentParameters`. In the latter case, it doesn't mean you have to specify everything in `ComponentParameters`. It just means we won't do any defaulting in that case and will use whatever value you provide, e.g. if memory requirement is missing, then we don't specify memory requirement in the deployment.

We would still need to do proper validation.

Thoughts?

Reply:
Thanks for following up! That would indeed have been a simpler approach and I think that’s the approach I was inclined to choose in the beginning. Since some considerable amount of development and testing effort was put into the current approach, I believe it’s best to keep with it, at least for now. I’ve been playing with it and believe that current version is actually pretty usable even for external users. The defaulting result will be visible on the next “kubectl edit” of BrokerCell, which I think makes it pretty discoverable. Documenting the process would further assist the experience. But at this point, it is not even clear if we want to expose this to external users, as the primary goal was to assist with the perf efforts. If we eventually decide to expose it externally and determine that the UX can be improved, we can have a follow up work item for that.

Reply:
As for the rest of suggestions, have accounted for them in the latest update.

Reply:
FYI, I have logged a task for the suggested approach: #1603. I'll follow up on this.